src/geo.c

```diff
@@ -50,21 +50,23 @@ int zslValueLteMax(double value, zrangespec *spec);
  * geoArray implementation
  * ==================================================================== */
 
-/* Create a new array of geoPoints. */
-geoArray *geoArrayCreate(void) {
-    geoArray *ga = zmalloc(sizeof(*ga));
-    /* It gets allocated on first geoArrayAppend() call. */
-    ga->array = NULL;
-    ga->buckets = 0;
+/* Init a new array of geoPoints. */
+void geoArrayInit(geoArray *ga) {
+    ga->buckets = MAX_GEO_ARRAY_BUFFER;
     ga->used = 0;
-    return ga;
+    ga->array = ga->arraybuf;
 }
 
 /* Add and populate with data a new entry to the geoArray. */
 geoPoint *geoArrayAppend(geoArray *ga, double *xy, double dist, double score, char *member) {
     if (ga->used == ga->buckets) {
-        ga->buckets = (ga->buckets == 0) ? 8 : ga->buckets * 2;
-        ga->array = zrealloc(ga->array, sizeof(geoPoint) * ga->buckets);
+        ga->buckets = ga->buckets * 2;
+        if (ga->array == ga->arraybuf) {
+            ga->array = zmalloc(sizeof(geoPoint) * ga->buckets);
+            memcpy(ga->array, ga->arraybuf, sizeof(geoPoint) * MAX_GEO_ARRAY_BUFFER);
```

Comment:
```suggestion
            memcpy(ga->array, ga->arraybuf, sizeof(*ga->arraybuf);
```

This is equivalent, if i'm not mistaken.